applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_prebuckling_solver.py

```diff
@@ -0,0 +1,110 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# Import applications
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+
+# Import base class file
+from KratosMultiphysics.StructuralMechanicsApplication.structural_mechanics_solver import MechanicalSolver
+
+from KratosMultiphysics import eigen_solver_factory
+from KratosMultiphysics import python_linear_solver_factory as linear_solver_factory
+import KratosMultiphysics.kratos_utilities as kratos_utils
+
+def CreateSolver(main_model_part, custom_settings):
+    return PrebucklingSolver(main_model_part, custom_settings)
+
+class PrebucklingSolver(MechanicalSolver):
+    """The structural mechanics prebuckling solver.
+
+    This class creates the mechanical solvers for prebuckling analysis.
+
+    See structural_mechanics_solver.py for more information.
+    """
+    def __init__(self, main_model_part, custom_settings):
+        # Construct the base solver.
+        super(PrebucklingSolver, self).__init__(main_model_part, custom_settings)
+        KratosMultiphysics.Logger.PrintInfo("::[PrebucklingSolver]:: ", "Construction finished")
+
+    @classmethod
+    def GetDefaultSettings(cls):
+        this_defaults = KratosMultiphysics.Parameters("""{
+            "buckling_settings"     : {
+                "initial_load_increment"    : 1.0,
+                "small_load_increment"      : 0.0005,
+                "path_following_step"       : 0.5,
+                "convergence_ratio"         : 0.05
+            },
+            "eigensolver_settings" : {
+                "solver_type"           : "eigen_eigensystem",
+                "max_iteration"         : 1000,
+                "tolerance"             : 1e-6,
+                "number_of_eigenvalues" : 5,
+                "echo_level"            : 1
+            }
+        }""")
+        this_defaults.AddMissingParameters(super(PrebucklingSolver, cls).GetDefaultSettings())
+        return this_defaults
+
+    #### Private functions ####
+    def AdvanceInTime(self, current_time):
```

Comment:
Because the new time depends on the computed buckling load. As the TIME parameter is considered to be the load multiplier.

Reply:
I spoke with @RiccardoRossi and I said that I think this is a bad idea. Many processes, utilities, strategies,... expect the time to have an specific meaning, this may lead to many unexpected behaviours and potential problems a the code may broke eventually. I think is better to define the load factor with another variable (for example LAMBDA) and modify the processes that compute the loads in order to allow another variable (for example "l"). In that case you don't need to consider "t" and the desired feature is preserved. Opinion @philbucher ?

Reply:
For the moment we have no better solution, so i think it should be accepted as it is.
If we eventually add the possibility of having another variable in the input functions to have this kind of behaviour we will eventually modify this, but i really would not do it in this PR (which is complex enough as of now)

Reply:
As you see I am not blocking, but I think the idea deserves consideration